modules/dnn/test/test_common.hpp

```diff
@@ -190,4 +231,45 @@ static inline bool readFileInMemory(const std::string& filename, std::string& co
     return true;
 }
 
+namespace opencv_test {
+
+using namespace cv::dnn;
+
+static testing::internal::ParamGenerator<tuple<Backend, Target> > dnnBackendsAndTargets(bool withInferenceEngine = true, bool withHalide = true)
```

Comment:
May be make `withHalide` is `false` by default? There are only two places where it's tested:

https://github.com/opencv/opencv/blob/3e027df5832ddb135f144d28b81c19bdf8a40133/modules/dnn/test/test_halide_layers.cpp#L47-L64

https://github.com/opencv/opencv/blob/3e027df5832ddb135f144d28b81c19bdf8a40133/modules/dnn/test/test_backends.cpp#L288-L301

Both have no `DNN_BACKEND_OPENCV, DNN_TARGET_CPU` because it is used as reference for outputs so I think we can apply this method something like this:
```cpp
dnnBackendsAndTargets(bool withInferenceEngine = true, bool withHalide = false, bool withCpuOCV = true)
```

Reply:
Done.